Source/WebKit/UIProcess/API/C/WKWebsiteDataRecordRef.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2023 Sony Interactive Entertainment Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "WKWebsiteDataRecordRef.h"
+
+#include "APIWebsiteDataRecord.h"
+#include "WKAPICast.h"
+#include "WKRetainPtr.h"
+#include "WKSecurityOriginRef.h"
+#include "WKString.h"
+
+WKTypeID WKWebsiteDataRecordGetTypeID()
+{
+    return WebKit::toAPI(API::WebsiteDataRecord::APIType);
+}
+
+WKWebsiteDataRecordRef WKWebsiteDataRecordCreate()
```

Comment:
This shouldn't exist.  Same with the setters.  Instead, I think we should make something like WKWebsiteDataStore.allWebsiteDataTypes, which will probably require making a WKSetRef type like WKDictionaryRef.

Reply:
Just kidding.  I meant fetchDataRecordsOfTypes, which means we don't need a WKSetRef.  We already have WKArrayRef.

Reply:
Does that mean that WKWebsiteDataRecordRef is unnecessary?

Reply:
It it probably necessary, but it should not be constructible or modifiable by API clients.  It is a read-only object that is only constructed inside of WebKit.